cpp/src/arrow/flight/sql/odbc/CMakeLists.txt

```diff
@@ -15,7 +15,76 @@
 # specific language governing permissions and limitations
 # under the License.
 
+# Use C++ 20 for ODBC and its subdirectory
+# GH-44792: Arrow will switch to C++ 20
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
 add_custom_target(arrow_flight_sql_odbc)
 
+# Ensure fmt is loaded as header only
+add_compile_definitions(FMT_HEADER_ONLY)
```

Comment:
Could you use `target_compile_definitions()` because `add_compile_definitions()`'s scope is directory? We should not use directory scope commands such as `add_compile_definitions()` and `include_directories()` for newly written code.